python/pyspark/pandas/frame.py

```diff
@@ -8828,22 +8842,108 @@ def describe(self, percentiles: Optional[List[float]] = None) -> "DataFrame":
         else:
             percentiles = [0.25, 0.5, 0.75]
 
-        formatted_perc = ["{:.0%}".format(p) for p in sorted(percentiles)]
-        stats = ["count", "mean", "stddev", "min", *formatted_perc, "max"]
+        if len(exprs_numeric) == 0:
+            if len(exprs_non_numeric) == 0:
+                raise ValueError("Cannot describe a DataFrame without columns")
 
-        sdf = self._internal.spark_frame.select(*exprs).summary(*stats)
-        sdf = sdf.replace("stddev", "std", subset=["summary"])
+            # Handling non-numeric type columns
+            # We will retrive the `count`, `unique`, `top` and `freq`.
+            sdf = self._internal.spark_frame.select(*exprs_non_numeric)
 
-        internal = InternalFrame(
-            spark_frame=sdf,
-            index_spark_columns=[scol_for(sdf, "summary")],
-            column_labels=column_labels,
-            data_spark_columns=[
-                scol_for(sdf, self._internal.spark_column_name_for(label))
-                for label in column_labels
-            ],
-        )
-        return DataFrame(internal).astype("float64")
+            # Get `count` & `unique` for each columns
+            has_timestamp_type = any(is_timestamp_types)
+            if not has_timestamp_type:
+                counts, uniques = map(
+                    lambda x: x[1:], sdf.summary("count", "count_distinct").take(2)
+                )
+            else:
+                # `summary` doesn't support for timestamp column, so we should manually compute it
+                # if timestamp type column exists.
+                counts = []
+                uniques = []
+                exprs = []
+                for column in exprs_non_numeric:
+                    exprs.append(F.count(column))
+                    exprs.append(F.count_distinct(column))
+
+                count_unique_values = sdf.select(*exprs).first()
+                for i in range(0, len(count_unique_values) - 1, 2):
+                    counts.append(str(count_unique_values[i]))
+                    uniques.append(str(count_unique_values[i + 1]))
+
+            # Get `top` & `freq` for each columns
+            tops = []
+            freqs = []
+            for column in exprs_non_numeric:
+                top, freq = sdf.groupby(column).count().sort("count", ascending=False).first()
+                tops.append(str(top))
+                freqs.append(str(freq))
+
+            stats = [counts, uniques, tops, freqs]
+            stats_names = ["count", "unique", "top", "freq"]
+
+            # Get `first` & `last` for each columns if timestamp type column exists.
+            if has_timestamp_type:
+                exprs = []
+                for is_timestamp_type, column, column_name in zip(
+                    is_timestamp_types, exprs_non_numeric, column_names
+                ):
+                    if is_timestamp_type:
+                        # `first` & `last` are min & max respectively for timestamp type.
```

Comment:
```
<stdin>:1: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
```

Looks like pandas will treat timestamps are numeric in the future releases. Can we just match the behaviour to the latest behaviour by default?

Reply:
Sounds good. Let me address!